File: transektcount/src/main/java/com/wmstein/transektcount/widgets/CountingWidget_head1.java
/*
 * Copyright (c) 2016. Wilhelm Stein, Bonn, Germany.
 */

package com.wmstein.transektcount.widgets;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.wmstein.transektcount.R;
import com.wmstein.transektcount.database.Count;

import androidx.annotation.NonNull;

/****************************************************
 * Interface for widget_counting_head1.xml
 * Created by wmstein 18.12.2016
 * Last edited on 2020-04-17
 */
public class CountingWidget_head1 extends ArrayAdapter<String>
{
    private String[] idArray;
    private String[] contentArray1;
    private String[] contentArray2;
    private String[] contentArray3;
    private Integer[] imageArray;

    public Count count;
    private LayoutInflater inflater;

    public CountingWidget_head1(Context context, int resource, String[] idArray, String[] nameArray, String[] nameArrayG, String[] codeArray, Integer[] imageArray)
    {
        super(context, resource, R.id.countName, nameArray);
        this.idArray = idArray;
        this.contentArray1 = nameArray;
        this.contentArray2 = codeArray;
        this.contentArray3 = nameArrayG;
        this.imageArray = imageArray;

        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View getDropDownView(int position, View convertView, @NonNull ViewGroup parent)
    {
        return getCustomView(position, parent);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent)
    {
        return getCustomView(position, parent);
    }

    private View getCustomView(int position, ViewGroup parent)
    {
        View head1 = inflater.inflate(R.layout.widget_counting_head1, parent, false);

        TextView countId = head1.findViewById(R.id.countId); // used for spinner interface  
        countId.setText(idArray[position]);

        TextView countName = head1.findViewById(R.id.countName);
        countName.setText(contentArray1[position]);

        TextView countNameg = head1.findViewById(R.id.countNameg);
        countNameg.setText(contentArray3[position]);

        TextView countCode = head1.findViewById(R.id.countCode);
        countCode.setText(contentArray2[position]);

        ImageView pSpecies = head1.findViewById(R.id.pSpecies);
        pSpecies.setImageResource(imageArray[position]);

        return head1;
    }

}

File: transektcount/src/main/java/com/wmstein/transektcount/widgets/CountingWidgetHead1.kt
/*
 * Copyright © 2016-2025. Wilhelm Stein, Bonn, Germany.
 */
package com.wmstein.transektcount.widgets

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.wmstein.transektcount.R

/*************************************************************
 * Interface for widget_counting_head1.xml
 * fills the species row with names, code, spinner and picture.
 * Created by wmstein 18.12.2016,
 * last edited in Java on 2023-05-09,
 * converted to Kotlin on 2023-06-26,
 * last edited on 2025-07-16
 */
class CountingWidgetHead1(
    context: Context,
    private val idArray: Array<String>,    // species ids
    private val nameArray: Array<String>,  // species names
    private val nameArrayL: Array<String>, // species local names
    private val codeArray: Array<String>,  // species codes
    private val imageArray: Array<Int>     // species images
) : ArrayAdapter<String?>(context, R.layout.widget_counting_head1, R.id.countName, nameArray) {

    private val inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    // Shows Spinner list
    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        return getCustomView(position, parent)
    }

    // Shows rest of counting page
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return getCustomView(position, parent)
    }

    private fun getCustomView(position: Int, parent: ViewGroup): View {
        val head1 = inflater.inflate(R.layout.widget_counting_head1, parent, false)
        val countId = head1.findViewById<TextView>(R.id.countId) // used for spinner interface  
        countId.text = idArray[position]
        val countName = head1.findViewById<TextView>(R.id.countName)
        countName.text = nameArray[position]
        val countNameg = head1.findViewById<TextView>(R.id.countNameg)
        countNameg.text = nameArrayL[position]
        val countCode = head1.findViewById<TextView>(R.id.countCode)
        countCode.text = codeArray[position]
        val pSpecies = head1.findViewById<ImageView>(R.id.pSpecies)
        pSpecies.setImageResource(imageArray[position])
        return head1
    }

}
